scripts/createTsIconComponents.js

```diff
@@ -0,0 +1,240 @@
+const fs = require('fs');
+const cheerio = require('cheerio');
+const camelcase = require('camelcase');
+const glob = require('glob');
+const path = require('path');
+const chalk = require('chalk');
+const rimraf = require('rimraf');
+const prettier = require('prettier');
+const prettierOptions = require('../.prettierrc');
+
+const rootDir = path.join(__dirname, '..', 'packages', 'ndla-icons');
+const attrs = ['xlink:href', 'id'];
+
+const copyright = fs.readFileSync(
+  path.join(__dirname, '..', 'COPYRIGHT'),
+  'utf-8',
+);
+const autoNotice = '// N.B! AUTOGENERATED FILE. DO NOT EDIT';
+
+function capitalize(string) {
+  return string.charAt(0).toUpperCase() + string.substring(1);
+}
+
+function ensureDirectoryExistence(filePath) {
+  const dirname = path.dirname(filePath);
+  if (fs.existsSync(dirname)) {
+    return;
+  }
+  ensureDirectoryExistence(dirname);
+  fs.mkdirSync(dirname);
+}
+
+function cleanAttributes($el, $) {
+  attrs.forEach(attr => {
+    $el.removeAttr(attr);
+  });
+  if ($el.children().length === 0) {
+    return false;
+  }
+
+  $el.children().each((index, el) => {
+    cleanAttributes($(el), $);
+  });
+  return true;
+}
+
+function camelCaseAttributes($el, $) {
+  Object.keys($el[0].attribs).forEach(attr => {
+    if (
+      $el[0].name !== 'svg' &&
+      attr.indexOf('-') > 0 &&
+      !attr.startsWith('data-')
+    ) {
+      const value = $el[0].attribs[attr];
+      $el.removeAttr(attr);
+      const camelCaseAttr = attr
+        .split('-')
+        .map((token, i) => (i === 0 ? token : capitalize(token)))
+        .join('');
+      $el.attr(camelCaseAttr, value);
+    }
+  });
+
+  if ($el.children().length === 0) {
+    return false;
+  }
+
+  $el.children().each((index, el) => {
+    camelCaseAttributes($(el), $);
+  });
+  return true;
+}
+
+function createComponent(name, svg) {
+  const $ = cheerio.load(svg, {
+    xmlMode: true,
+  });
+  const $svg = $('svg');
+  const viewBox = $svg.attr('viewBox');
+  const license = $svg.attr('data-license');
+  const source = $svg.attr('data-source');
+  if (!license || !source) {
+    return undefined;
+  }
+
+  cleanAttributes($svg, $);
+  camelCaseAttributes($svg, $);
+
+  const iconSvg = $svg.html();
+  return prettier.format(
+    `${copyright}
+${autoNotice}
+import React, { ReactNode } from 'react';
+import Icon from '../Icon';
+
+interface Props {
+  children?: ReactNode;
+}
```

Comment:
er dette en .ts/tsx fil eller skal den være .js?

Reply:
Spiller vel egentlig ingen rolle siden den bare kjøres som et script for å generere filene som så skal brukes under kjøringa